packages/channels/botpress-channel-web/src/api.js

```diff
@@ -191,15 +191,13 @@ module.exports = async (bp, config) => {
       return res.status(400).send(ERR_USER_ID_REQ)
     }
 
-    await getOrCreateUser(userId) // Just to create the user if it doesn't exist
-
-    const conversations = await listConversations(userId)
+    await ensureUserExists(userId)
 
-    return res.send([...conversations])
+    return res.send(await listConversations(userId))
   })
 
   function validateUserId(userId) {
-    return /[a-z0-9-_]+/i.test(userId)
+    return /^[a-z0-9-_]+$/i.test(userId)
```

Comment:
idea: if this is called for every incoming message we could extract the regex, which would compile it only once